lib/ansible/modules/network/eos/eos_l2_interface.py

```diff
@@ -0,0 +1,298 @@
+                        if trunk_allowed_vlans:
+                            commands.append('switchport trunk allowed vlan {0}'.format(trunk_allowed_vlans))
+                    else:
+                        if obj_in_have['native_vlan']:
+                            commands.append('not switchport trunk native vlan {}'.format(obj_in_have['native_vlan']))
+                        if obj_in_have['trunk_allowed_vlans']:
+                            commands.append('not switchport trunk allowed vlan {}'.format(obj_in_have['trunk_allowed_vlans']))
+                        commands.append('switchport access vlan {0}'.format(access_vlan))
+                else:
+                    if mode == 'access':
+                        if access_vlan != obj_in_have['access_vlan']:
+                            commands.append('switchport access vlan {0}'.format(access_vlan))
+                    else:
+                        if native_vlan != obj_in_have['native_vlan'] and native_vlan:
+                            commands.append('switchport trunk native vlan {0}'.format(native_vlan))
+                        if trunk_allowed_vlans != obj_in_have['trunk_allowed_vlans'] and trunk_allowed_vlans:
+                            commands.append('switchport trunk allowed vlan {0}'.format(trunk_allowed_vlans))
+
+        if commands[-1] == interface:
+            commands.pop(-1)
+
+    return commands
+
+
+def map_config_to_obj(module):
+    config = get_config(module, flags=['| section interface'])
+    configobj = NetworkConfig(indent=3, contents=config)
+
+    match = re.findall(r'^interface (\S+)', config, re.M)
+    if not match:
+        return list()
+
+    instances = list()
+
+    for item in set(match):
+        command = 'sh int {0} switchport | include Switchport'
+        switchport_cfg = run_commands(module, command.format(item))[0].split(':')[1].strip()
+        if switchport_cfg == 'Enabled':
+            state = 'present'
+        else:
+            state = 'absent'
+
+        obj = {
+            'name': item.lower(),
+            'state': state,
+        }
+
+        if state == 'present':
+            obj['access_vlan'] = parse_config_argument(configobj, item, 'switchport access vlan')
+            obj['native_vlan'] = parse_config_argument(configobj, item, 'switchport trunk native vlan')
+            obj['trunk_allowed_vlans'] = parse_config_argument(configobj, item, 'switchport trunk allowed vlan')
+            if obj['access_vlan']:
+                obj['mode'] = 'access'
+            else:
+                obj['mode'] = 'trunk'
+
+        instances.append(obj)
+
+    return instances
+
+
+def map_params_to_obj(module):
+    obj = []
+
+    aggregate = module.params.get('aggregate')
+    if aggregate:
+        for item in aggregate:
+            for key in item:
+                if item.get(key) is None:
+                    item[key] = module.params[key]
+
+            item['name'] = item['name'].lower()
```

Comment:
Should this handle for space within interface name?
Something similar to  https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/eos/eos_vrf.py#L224